application/controllers/artifacts_controller.py

```diff
@@ -33,27 +34,32 @@ def show(self, object_id):
     def index(self, params):
         """Logic for querying several artifacts"""
 
-        artifacts = Artifact.search(params)
+        artifacts = ElasticArtifact.search(params)
 
         return {"images": respond_with(artifacts)}, 200
 
     @use_args(artifacts_validator.create_args())
     def create(self, params):
         """Logic for creating an artifact"""
-        params = self.upload_file(params)
-        artifact = Artifact(params)
+        metadata = type(self)._upload_file(params)
+        artifact = type(self)._create_artifact(params, metadata)
+        ImageRecognizer.auto_add_tags(artifact.elastic)
+        return respond_with(artifact.elastic), 200
```

Comment:
Split into two methods and calling the wrapper instead of ElasticArtifact, everywhere that modifies values to keep both data models up to date. For now ElasticSearch will be the one used primarily though.